### PR TITLE
introduce Solver Ingress Mode

### PR DESCRIPTION
**Pull request by nahimterrazas**
# Summary

## Checklist

- [ ] Add a reference to related issues in the PR description.
- [ ] Add unit tests if applicable.

## Summary by CodeRabbit

* **New Features**
  * Added SOLVER_INGRESS_MODE to control solver intake: `active` (default) or `intake_disabled`, which stops accepting new quote requests, order submissions, and discovered intents while preserving existing processing and admin operations.

* **Behavior Changes**
  * When intake is disabled, submission endpoints immediately return 503 Service Unavailable with a clear "SOLVER_INTAKE_DISABLED" error.

* **Documentation**
  * Updated docs and Docker environment section with usage and restart guidance for the new mode.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @NicoMolinaOZ @nahimterrazas @pepebndc @shahnami
\ No newline at end of file
+* @nahimterrazas @pepebndc @shahnami
```
Comment (tirumerla) [inline .github/CODEOWNERS]: nit:EOL

File: README.md [hunk 2]
```diff
@@ -392,9 +393,25 @@ Legacy alias support:
 | `REDIS_URL` | Yes | Redis connection URL (default: `redis://localhost:6379`). For production, use managed services like AWS ElastiCache |
 | `SOLVER_PRIVATE_KEY` | Conditional | 64-character hex private key (without 0x prefix) |
 | `SOLVER_ID` | For loading | Solver ID to load from Redis (set after first seed) |
+| `SOLVER_INGRESS_MODE` | No | `active` by default; set `intake_disabled` to stop accepting new solver work |
```
Comment (tirumerla) [inline README.md]: Can we have one set for defining env. variables. Having one call out specific to docker and one call out specific to running locally causes confusion. May be it's better to call out env variables specifically needed for docker vs just running local. Just a nit.

File: crates/solver-service/src/validators/intake.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	Ok(())
+}
+
+pub fn intake_disabled_api_error() -> APIError {
+	APIError::ServiceUnavailable {
+		error_type: ApiErrorType::SolverIntakeDisabled,
+		message: "Solver intake is disabled; new quotes and orders are temporarily unavailable"
+			.to_string(),
+		retry_after: None,
+	}
```
Comment (tirumerla) [inline crates/solver-service/src/validators/intake.rs]: You can use `QuoteError::SolverIntakeDisabled.into()` instead of deduping msg

File: crates/solver-service/src/config_merge.rs [hunk 4]
```diff
@@ -3115,6 +3145,49 @@ mod tests {
...
+	#[test]
+	fn test_parse_solver_ingress_mode_accepts_intake_disabled() {
+		assert_eq!(
+			parse_solver_ingress_mode("intake_disabled").unwrap(),
+			solver_config::SolverIngressMode::IntakeDisabled
+		);
+	}
+
+	#[test]
+	fn test_parse_solver_ingress_mode_rejects_unknown_value() {
```
Comment (tirumerla) [inline crates/solver-service/src/config_merge.rs]: nit: Let's add a test with empty and whitespace just in case to catch those cases to make sure it's still active.

File: crates/solver-core/src/handlers/intent.rs [hunk 5]
```diff
@@ -140,6 +140,20 @@ impl IntentHandler {
 		// This enables hot-reload: config changes via Admin API are picked up on next intent.
 		let config = self.dynamic_config.read().await.clone();
 
+		if config.solver.is_intake_disabled() {
```
Comment (tirumerla) [inline crates/solver-core/src/handlers/intent.rs:145]: Let's add a comment to mention this is a backstop for intake disabled

File: README.md [hunk 6]
```diff
@@ -392,9 +393,25 @@ Legacy alias support:
...
+When enabled, the solver rejects:
+
+- `POST /api/v1/quotes`
+- `POST /api/v1/orders` quote acceptances
+- `POST /api/v1/orders` direct submissions
+- newly discovered on-chain/off-chain intents
+
+The solver continues processing already accepted orders and settlement events. Health checks, asset reads, admin balance reads, and admin withdrawals remain available.
+
+Use this for custody-cap operations where OpenZeppelin must stop new intake without receiving Polygon admin withdrawal permissions. Restore normal operation by restarting/redeploying with `SOLVER_INGRESS_MODE=active` or with the variable unset.
```
Comment (tirumerla) [inline README.md]:
```suggestion
Use this for custody-cap operations where OpenZeppelin must stop new intake without receiving Polygon admin withdrawal permissions. 

**Toggling the mode requires a process restart/redeploy.** 
Restore normal operations by setting `SOLVER_INGRESS_MODE=active` or with the variable unset which requires restarting the process.
```

File: crates/solver-config/src/lib.rs [hunk 7]
```diff
@@ -102,6 +102,32 @@ pub struct SolverConfig {
...
+}
+
+impl SolverIngressMode {
+	pub fn is_intake_disabled(self) -> bool {
+		matches!(self, Self::IntakeDisabled)
+	}
+}
+
+impl SolverConfig {
+	pub fn is_intake_disabled(&self) -> bool {
```
Comment (tirumerla) [inline crates/solver-config/src/lib.rs:128]: nit: can we make it consistent using `&self`

**Review by tirumerla — APPROVED** [review]
Lgtm, added some nits and couple comments

File: crates/solver-service/src/validators/intake.rs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+	Ok(())
+}
+
+pub fn ensure_order_intake_enabled(config: &Config) -> Result<(), APIError> {
+	if config.solver.is_intake_disabled() {
+		return Err(QuoteError::SolverIntakeDisabled.into());
+	}
+	Ok(())
+}
```
Comment (shahnami) [inline crates/solver-service/src/validators/intake.rs]: Can this not be a single _more generic_ function instead? They do the same thing?

File: crates/solver-service/src/config_merge.rs [hunk 9]
```diff
@@ -1123,6 +1117,42 @@ pub fn build_runtime_config(operator_config: &OperatorConfig) -> Result<Config,
...
 
+fn parse_solver_ingress_mode(raw: &str) -> Result<SolverIngressMode, MergeError> {
+	match raw.trim().to_ascii_lowercase().as_str() {
+		"" | "active" => Ok(SolverIngressMode::Active),
+		"intake_disabled" => Ok(SolverIngressMode::IntakeDisabled),
+		other => Err(MergeError::Validation(format!(
+			"Invalid SOLVER_INGRESS_MODE value '{other}'. Expected 'active' or 'intake_disabled'"
+		))),
+	}
+}
```
Comment (shahnami) [inline crates/solver-service/src/config_merge.rs]: Can this not be a boolean?

**Review by shahnami — APPROVED** [review]
This is fine for now, but I feel like this can be more easily achieved with disabling API capabilities at a single layer?

**Comment by nahimterrazas** [comment]
> This is fine for now, but I feel like this can be more easily achieved with disabling API capabilities at a single layer?

This can be centralized for the HTTP routes, but I don’t think a single API layer fully covers the feature. `intake_disabled` needs to stop all new solver work, and discovered intents go through `solver-core::IntentHandler::handle`, not through the API router. An API-only capability check would block `/quotes` and `/orders`, but listeners could still process newly discovered intents.
